plugins/modules/proxmox_cluster.py

```diff
@@ -0,0 +1,302 @@
+    link0 = module.params.get('link0')
+    link1 = module.params.get('link1')
+
+    if link0:
+        payload['link0'] = link0
+    if link1:
+        payload['link1'] = link1
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Cluster '{}' would be created (check mode).".format(cluster_name), cluster=cluster_name)
+
+    try:
+        proxmox.cluster.config.post(**payload)
+    except Exception as e:
+        module.fail_json(msg="Error while creating cluster: {}".format(str(e)))
+
+    module.exit_json(changed=True, msg="Cluster '{}' created.".format(cluster_name), cluster=cluster_name)
+
+
+def get_cluster_info(proxmox, module):
+    try:
+        data = proxmox.cluster.config.join.get()
+    except Exception as e:
+        module.fail_json(msg="Error while obtaining cluster join information: {}".format(str(e)))
+
+    module.exit_json(changed=False, ansible_facts={'cluster_join_info': data})
+
+
+def join_cluster(proxmox, module):
+    master_ip = module.params['master_ip']
+    fingerprint = module.params['fingerprint']
+    api_password = module.params['api_password']
+    cluster_name = module.params.get('cluster_name')
+
+    if not master_ip or not fingerprint:
+        module.fail_json(msg="Action 'join_cluster' requires 'master_ip' and 'fingerprint' arguments.")
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Node would join the cluster (check mode).", cluster=cluster_name)
+
+    try:
+        proxmox.cluster.config.join.post(
+            hostname=master_ip,
+            fingerprint=fingerprint,
+            password=api_password
+        )
+
+    except Exception as e:
+        module.fail_json(msg="Error while joining the cluster: {}".format(str(e)))
+
+    module.exit_json(changed=True, msg="Node joined the cluster.", cluster=cluster_name)
+
+
+def main():
+    module_args = dict(
+        action=dict(type='str', required=True, choices=['create_cluster', 'get_cluster_info', 'join_cluster']),
+        api_host=dict(type='str', required=True),
+        api_user=dict(type='str', required=True),
+        api_password=dict(type='str', required=True, no_log=True),
+        api_ssl_verify=dict(type='bool', required=False, default=True),
+        cluster_name=dict(type='str', required=False, default=None),
+        link0=dict(type='str', required=False, default=None),
+        link1=dict(type='str', required=False, default=None),
+        master_ip=dict(type='str', required=False),
+        fingerprint=dict(type='str', required=False),
+    )
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    if __MISSING_MODULE__:
+        module.fail_json(msg="Required Python module is missing: proxmoxer")
+
+    action = module.params['action']
+    api_host = module.params['api_host']
+    api_user = module.params['api_user']
+    api_password = module.params['api_password']
+    api_ssl_verify = module.params['api_ssl_verify']
```

Comment:
probably better to use `module.params.get(...)` here too

Reply:
ack, let's streamline it